File: Code/Java/ExperimentoRemoto/src/Timer.java
/*
 * To change this template, choose Tools | Templates
 * and open the template in the editor.
 */

/**
 *
 * @author Henrique
 */
public class Timer {
    
    private int tempo;
    private long lastnow;
    private boolean ligado = false;
    
    public Timer(int t){
        tempo = t;
    }
    public void Set(int t){
        tempo = t;
    }
    
    public void Start(){
        ligado = true;
        lastnow = System.currentTimeMillis();
    }
    
    public void Stop(){
        ligado = false;
    }
    
    public boolean Get(){
        if (!ligado) return false;
        long now = System.currentTimeMillis();
        long total = now - lastnow;
        if (total >= tempo){
            lastnow = now;
            return true;
        }
        return false;
    }
}

File: Code/Arduino/libraries/Timer/Timer.cpp
/*

  Timer.cpp 
  
  V1.0
  
  14/09/2011
  
  Luiz Henrique Cassettari
  
*/

#include "Timer.h"

Timer::Timer(unsigned long ms){
  ligado = false;
  tempo = ms;
  lastnow = 0;
}

void Timer::Set(unsigned long ms){
  tempo = ms;
}

boolean Timer::Get(){
  if (!ligado) return false;
  unsigned long now = millis();
  unsigned long total = now - lastnow;
  if (total >= tempo){
    lastnow = now;
    return true;
  }
  return false;
}

void Timer::Start(){
  ligado = true;
  lastnow = millis();
}

void Timer::Stop(){
  ligado = false;
}


